libs/wingsdk/src/shared-gcp/bucket.inflight.ts

```diff
@@ -47,20 +49,34 @@ export class BucketClient implements IBucketClient {
     }
   }
 
-  public async put(key: string, body: string): Promise<void> {
-    try {
-      await this.bucket.file(key).save(body);
-    } catch (error) {
-      throw new Error(`Failed to put object. (key=${key})`);
-    }
+  /**
+   * Put object into bucket with given body contents
+   *
+   * @param key Key of the object
+   * @param body string contents of the object
+   */
+  public async put(
+    key: string,
+    body: string,
+    opts?: BucketPutOptions
+  ): Promise<void> {
+    const options = {
+      contentType:
+        (opts?.contentType ?? mime.lookup(key)) || "application/octet-stream",
+    };
+    await this.bucket.file(key).save(body, options);
```

Comment:
Can we add update the unit tests [here](https://github.com/winglang/wing/blob/main/libs/wingsdk/test/shared-gcp/bucket.inflight.test.ts)?

I think we essentially want a test that checks if given someone called `put()` in our `Bucket` class (the one defined in this file) without specifying a `contentType`, then `save()` should have been called with "application/octet-stream" in its options.

Reply:
Yeah, I think that makes sense. I honestly was surprised to see that the Node.js SDK for GCP automatically infers the type for known extensions, but not for unknown ones. Yet, if you upload files manually to Cloud Storage from the Google Cloud Console, it automatically sets the type to `application/octet-stream` if it doesn't recognize the file type.

Reply:
Yeah, I think that makes sense. I honestly was surprised to see that the Node.js SDK for GCP automatically sets the type for known extensions, but not for unknown ones. Yet, if you upload files manually to Cloud Storage from the Google Cloud Console, it automatically sets the type to `application/octet-stream` if it doesn't recognize the file type.

Reply:
Apparently you cannot `vi.spyOn()` on the methods that are called internally by `BucketClient`, so the only solution I see for adding the test would be to merge #4633 so that you can call `put()` followed by `metadata()` to check that the Content-Type is indeed `application/octet-stream` when uploading an object with unknown extension.

Reply:
Sounds strange,
is it because they are private? (you can easily bypass by `@ts-expect-error`, there are other solutions on the internet as well)
or there is another problem?